python/ray/rllib/agents/ppo/appo_policy.py

```diff
@@ -357,6 +398,25 @@ def clip_gradients(policy, optimizer, loss):
     clipped_grads = list(zip(policy.grads, policy.var_list))
     return clipped_grads
 
+class KLCoeffMixin(object):
+    def __init__(self, config):
+        # KL Coefficient
+        self.kl_coeff_val = config["kl_coeff"]
+        self.kl_target = config["kl_target"]
+        self.kl_coeff = tf.get_variable(
+            initializer=tf.constant_initializer(self.kl_coeff_val),
+            name="kl_coeff",
+            shape=(),
+            trainable=False,
+            dtype=tf.float32)
+
+    def update_kl(self, sampled_kl):
+        if sampled_kl > 2.0 * self.kl_target:
+            self.kl_coeff_val *= 1.5
+        elif sampled_kl < 0.5 * self.kl_target:
+            self.kl_coeff_val *= 0.5
+        self.kl_coeff.load(self.kl_coeff_val, session=self.get_session())
+        return self.kl_coeff_val
```

Comment:
would it be so bad just to import this from ppo_policy?

Reply:
I'll try it. Believe it is just importing another Mixin from ppo_policy